src/Compilers/Core/Portable/Diagnostic/CustomObsoleteDiagnosticInfo.cs

```diff
@@ -50,6 +56,11 @@ public override DiagnosticDescriptor Descriptor
             }
         }
 
+        internal override DiagnosticInfo GetInstanceWithSeverity(DiagnosticSeverity severity)
```

Comment:
Did you look at the other types that derive from `DiagnosticInfo` to see if they needed to also override this?

Reply:
This is the only one because it's the only one that overrides Descriptor and MessageIdentifier.